### PR TITLE
[NOD-1130] Integrate RPC with the new architecture

### PR DESCRIPTION
**Pull request by stasatdaglabs**

### CODE REVIEW

File: cmd/kaspaminer/mineloop.go [hunk 1]
```diff
@@ -119,22 +119,22 @@ func solveBlock(block *util.Block, stopChan chan struct{}, foundBlock chan *util
 
 }
 
-func templatesLoop(client *minerClient, miningAddr util.Address,
-	newTemplateChan chan *rpcmodel.GetBlockTemplateResult, errChan chan error, stopChan chan struct{}) {
+func templatesLoop(minerClient *minerClient, miningAddr util.Address,
```
Comment (someone235) [inline cmd/kaspaminer/mineloop.go:122]: Can you rename back to client to avoid shadowing?

File: protocol/blocks.go [hunk 2]
```diff
@@ -39,5 +38,11 @@ func (m *Manager) OnNewBlock(block *util.Block) error {
 	copy(txIDsToBroadcast[len(transactionsAcceptedToMempool):], txIDsToBroadcast)
 	txIDsToBroadcast = txIDsToBroadcast[:wire.MaxInvPerTxInvMsg]
 	inv := wire.NewMsgTxInv(txIDsToBroadcast)
-	return m.netAdapter.Broadcast(peerpkg.ReadyPeerIDs(), inv)
+	return m.netAdapter.Broadcast(m.peers.ReadyPeerIDs(), inv)
+}
+
+// AddBlock adds the given block to the DAG and propagates it.
+func (m *Manager) AddBlock(block *util.Block) error {
+	// TODO(libp2p): unimplemented
```
Comment (someone235) [inline protocol/blocks.go:46]: Please create ticket, since it's core functionality

File: rpc/handle_send_raw_transaction.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	"github.com/kaspanet/kaspad/wire"
+	"github.com/pkg/errors"
+)
+
+// handleSendRawTransaction implements the sendRawTransaction command.
+func handleSendRawTransaction(s *Server, cmd interface{}, closeChan <-chan struct{}) (interface{}, error) {
+	c := cmd.(*model.SendRawTransactionCmd)
+	// Deserialize and send off to tx relay
+	hexStr := c.HexTx
+	if len(hexStr)%2 != 0 {
```
Comment (someone235) [inline rpc/handle_send_raw_transaction.go]: Please get rid of this PHP behaviour

File: rpc/model/rpc_results.go [hunk 4]
```diff
@@ -219,25 +205,11 @@ type GetNetworkInfoResult struct {
...
-	LastRecv    int64   `json:"lastRecv"`
-	BytesSent   uint64  `json:"bytesSent"`
-	BytesRecv   uint64  `json:"bytesRecv"`
-	ConnTime    int64   `json:"connTime"`
-	TimeOffset  int64   `json:"timeOffset"`
-	PingTime    float64 `json:"pingTime"`
-	PingWait    float64 `json:"pingWait,omitempty"`
-	Version     uint32  `json:"version"`
-	SubVer      string  `json:"subVer"`
-	Inbound     bool    `json:"inbound"`
```
Comment (someone235) [inline rpc/model/rpc_results.go:236]: I would like to keep it

File: rpc/model/rpc_results.go [hunk 5]
```diff
@@ -219,25 +205,11 @@ type GetNetworkInfoResult struct {
...
-	SelectedTip string  `json:"selectedTip,omitempty"`
-	BanScore    int32   `json:"banScore"`
```
Comment (someone235) [inline rpc/model/rpc_results.go:238]: I would like to keep it

File: rpc/model/rpc_results.go [hunk 6]
```diff
@@ -219,25 +205,11 @@ type GetNetworkInfoResult struct {
...
-	ID          int32   `json:"id"`
-	Addr        string  `json:"addr"`
-	Services    string  `json:"services"`
-	RelayTxes   bool    `json:"relayTxes"`
-	LastSend    int64   `json:"lastSend"`
-	LastRecv    int64   `json:"lastRecv"`
-	BytesSent   uint64  `json:"bytesSent"`
-	BytesRecv   uint64  `json:"bytesRecv"`
-	ConnTime    int64   `json:"connTime"`
-	TimeOffset  int64   `json:"timeOffset"`
```
Comment (someone235) [inline rpc/model/rpc_results.go:231]: I would like to keep it

File: rpc/model/rpc_results.go [hunk 7]
```diff
@@ -219,25 +205,11 @@ type GetNetworkInfoResult struct {
...
-	PingTime    float64 `json:"pingTime"`
-	PingWait    float64 `json:"pingWait,omitempty"`
-	Version     uint32  `json:"version"`
-	SubVer      string  `json:"subVer"`
```
Comment (someone235) [inline rpc/model/rpc_results.go:235]: I would like to keep it

File: rpc/model/rpc_results.go [hunk 8]
```diff
@@ -219,25 +205,11 @@ type GetNetworkInfoResult struct {
...
-	RelayTxes   bool    `json:"relayTxes"`
-	LastSend    int64   `json:"lastSend"`
-	LastRecv    int64   `json:"lastRecv"`
-	BytesSent   uint64  `json:"bytesSent"`
-	BytesRecv   uint64  `json:"bytesRecv"`
-	ConnTime    int64   `json:"connTime"`
-	TimeOffset  int64   `json:"timeOffset"`
-	PingTime    float64 `json:"pingTime"`
-	PingWait    float64 `json:"pingWait,omitempty"`
-	Version     uint32  `json:"version"`
```
Comment (someone235) [inline rpc/model/rpc_results.go:234]: I would like to keep it

File: rpc/model/rpc_results.go [hunk 9]
```diff
@@ -219,25 +205,11 @@ type GetNetworkInfoResult struct {
...
 type GetConnectedPeerInfoResult struct {
-	ID          int32   `json:"id"`
-	Addr        string  `json:"addr"`
-	Services    string  `json:"services"`
-	RelayTxes   bool    `json:"relayTxes"`
-	LastSend    int64   `json:"lastSend"`
-	LastRecv    int64   `json:"lastRecv"`
-	BytesSent   uint64  `json:"bytesSent"`
-	BytesRecv   uint64  `json:"bytesRecv"`
-	ConnTime    int64   `json:"connTime"`
```
Comment (someone235) [inline rpc/model/rpc_results.go:230]: I would like to keep it

File: protocol/blocks.go (hunk 2 above)
Comment (svarogg) [reply protocol/blocks.go:46]: This is beyond "core functionality" - this+getBlockTemplate are the first and foremost rpc calls we are going to use in tests, without them there's little point to RPC.

I'd rather if you implemented this here, or at the very least - this should be the next ticket you do.

File: rpc/handle_connect.go [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+import (
+	"github.com/kaspanet/kaspad/rpc/model"
+	"github.com/kaspanet/kaspad/util/network"
+)
+
+// handleConnect handles connect commands.
+func handleConnect(s *Server, cmd interface{}, closeChan <-chan struct{}) (interface{}, error) {
+	c := cmd.(*model.ConnectCmd)
+
+	oneTry := c.OneTry != nil && *c.OneTry
```
Comment (svarogg) [inline rpc/handle_connect.go:12]: How about we change OneTry to IsPermanent?

File: rpc/handle_get_block_template.go [hunk 11]
```diff
@@ -770,3 +770,21 @@ func decodeLongPollID(longPollID string) ([]*daghash.Hash, int64, error) {
...
 }
+
+// isSynced checks if the node is synced enough based upon its worldview.
+// This is used to determine if the node can support mining and requesting newly-mined blocks.
+// To do that, first it checks if the selected tip timestamp is not older than maxTipAge. If that's the case, it means
+// the node is synced since blocks' timestamps are not allowed to deviate too much into the future.
+// If that's not the case it checks the rate it added new blocks to the DAG recently. If it's faster than
+// blockRate * maxSyncRateDeviation it means the node is not synced, since when the node is synced it shouldn't add
+// blocks to the DAG faster than the block rate.
+func isSynced(s *Server) bool {
```
Comment (svarogg) [inline rpc/handle_get_block_template.go:781]: This is very business-logicy, shouldn't be in the RPC package.
Please either move to blockdag (or wherever relavant), or create a ticket

File: protocol/blocks.go (hunk 2 above)
Comment (stasatdaglabs) [reply protocol/blocks.go:46]: Created.

File: rpc/handle_get_connected_peer_info.go [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+import (
+	"github.com/kaspanet/kaspad/rpc/model"
+)
+
+// handleGetConnectedPeerInfo implements the getConnectedPeerInfo command.
+func handleGetConnectedPeerInfo(s *Server, cmd interface{}, closeChan <-chan struct{}) (interface{}, error) {
+	peers := s.protocolManager.Peers()
+	infos := make([]*model.GetConnectedPeerInfoResult, 0, len(peers))
+	for _, peer := range peers {
+		info := &model.GetConnectedPeerInfoResult{
```
Comment (svarogg) [inline rpc/handle_get_connected_peer_info.go:12]: Please create ticket if there are any fields that are left TODO.

File: rpc/handle_get_info.go [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+
+import (
+	"github.com/kaspanet/kaspad/rpc/model"
+	"github.com/kaspanet/kaspad/version"
+)
+
+// handleGetInfo implements the getInfo command. We only return the fields
+// that are not related to wallet functionality.
+func handleGetInfo(s *Server, cmd interface{}, closeChan <-chan struct{}) (interface{}, error) {
+	ret := &model.InfoDAGResult{
```
Comment (svarogg) [inline rpc/handle_get_info.go:11]: If there are any fields that are left TODO - please create a ticket

File: rpc/handle_get_net_totals.go [hunk 14]
```diff
@@ -1,16 +1,16 @@
...
 	"time"
 )
 
 // handleGetNetTotals implements the getNetTotals command.
 func handleGetNetTotals(s *Server, cmd interface{}, closeChan <-chan struct{}) (interface{}, error) {
-	totalBytesRecv, totalBytesSent := s.cfg.ConnMgr.NetTotals()
-	reply := &rpcmodel.GetNetTotalsResult{
-		TotalBytesRecv: totalBytesRecv,
-		TotalBytesSent: totalBytesSent,
+	// TODO(libp2p): fill this up with real values
```
Comment (svarogg) [inline rpc/handle_get_net_totals.go:10]: Please create ticket if you haven't yet.

File: rpc/handle_get_peer_addresses.go [hunk 15]
```diff
@@ -1,26 +1,26 @@
...
 // handleGetPeerAddresses handles getPeerAddresses commands.
 func handleGetPeerAddresses(s *Server, cmd interface{}, closeChan <-chan struct{}) (interface{}, error) {
-	peersState, err := s.cfg.addressManager.PeersStateForSerialization()
+	peersState, err := s.addressManager.PeersStateForSerialization()
 	if err != nil {
 		return nil, err
 	}
 
-	rpcPeersState := rpcmodel.GetPeerAddressesResult{
+	rpcPeersState := model.GetPeerAddressesResult{
```
Comment (svarogg) [inline rpc/handle_get_peer_addresses.go:12]: Please create ticket if there are any fields that are left TODO.

File: rpc/handle_get_connected_peer_info.go (hunk 12 above)
Comment (stasatdaglabs) [reply rpc/handle_get_connected_peer_info.go:12]: Done.

File: rpc/handle_get_info.go (hunk 13 above)
Comment (stasatdaglabs) [reply rpc/handle_get_info.go:11]: There aren't.

File: rpc/handle_get_net_totals.go (hunk 14 above)
Comment (stasatdaglabs) [reply rpc/handle_get_net_totals.go:10]: Created.

File: rpc/handle_get_peer_addresses.go (hunk 15 above)
Comment (stasatdaglabs) [reply rpc/handle_get_peer_addresses.go:12]: There aren't.

**Pull request merged**